File: python_leaoai/repositories/Implementations/Factory/EmbalagemFactory.py
# repositories/embalagem_factory.py
from sqlalchemy.orm import Session
from typing import List, Optional
from models.factory import Embalagem

class EmbalagemFactory:
    def __init__(self, db: Session):
        self.db = db

    # --- CRUD ---
    def get_obj(self, id: int) -> Optional[Embalagem]:
        return self.db.query(Embalagem).filter(Embalagem.id == id).first()

    def save_obj(self, obj: Embalagem) -> Embalagem:
        self.db.add(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def update_obj(self, obj: Embalagem) -> Embalagem:
        self.db.merge(obj)
        self.db.commit()
        self.db.refresh(obj)
        return obj

    def get_all(self) -> List[Embalagem]:
        return self.db.query(Embalagem).all()

    def get_list_actives(self) -> List[Embalagem]:
        return self.db.query(Embalagem).filter(Embalagem.ativo == True).all()

    def get_list(self) -> List[Embalagem]:
        return self.db.query(Embalagem).filter(Embalagem.excluido == False).join(Embalagem.embalagem_type).all()

    def delete_obj(self, id: int) -> bool:
        obj = self.get_obj(id)
        if not obj:
            return False
        try:
            obj.ativo = False
            obj.excluido = True
            self.db.commit()
            return True
        except:
            self.db.rollback()
            return False

File: Framework.Repositories/Implementations/Factory/EmbalagemFactory.cs
using Data;
using Framework.Data.Models.Factory;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Linq;

namespace Framework.Repositories.Implementations.Factory
{
    public class EmbalagemFactory
    {
        private readonly ApplicationDbContext _context;

        public EmbalagemFactory(ApplicationDbContext context)
        {
            _context = context;
        }

        #region Crud
        public Embalagem GetObj(int id)
        {
            return _context.Embalagems.FirstOrDefault(x => x.Id == id);
        }

        public Embalagem SaveObj(Embalagem obj)
        {
            _context.Embalagems.Add(obj);
            _context.SaveChanges();
            return obj;
        }

        public Embalagem UpdateObj(Embalagem obj)
        {
            _context.Embalagems.Update(obj);
            _context.SaveChanges();
            return obj;
        }

        public List<Embalagem> GetAll()
        {
            return _context.Embalagems.ToList();
        }
        public List<Embalagem> GetListActives()
        {
            return _context.Embalagems.Where(x => x.Ativo).ToList();
        }
        public List<Embalagem> GetList()
        {
            return _context.Embalagems.Where(x => !x.Excluido).Include(b=>b.EmbalagemType).ToList();
        }

        public bool DeleteObj(int id)
        {
            var obj = GetObj(id);
            if (obj == null)
                return false;
            try
            {
                obj.Ativo = false;
                obj.Excluido = true;
                _context.Embalagems.Update(obj);
                _context.SaveChanges();
                return true;
            }
            catch (Exception)
            {
                return false;
            }
        }
        #endregion
    }
}
